fendermint/app/src/validators.rs

```diff
@@ -66,4 +63,28 @@ impl<C: Client + Sync> ValidatorTracker<C> {
         let keys = self.public_keys.read().unwrap();
         keys.get(id).copied()
     }
+
+    pub fn hydrate_cache(
```

Comment:
I assume you added this to prevent `ValidatorTracker` from dispatching calls during `begin_block`. If so, it would be worthwhile to refactor it, as it should no longer use `self.client` at all, and should panic if a validator record is missing.